### PR TITLE
PLATUI-2380 use LazyList instead of deprecated Stream

### PR DESCRIPTION
**Pull request by ellamdav**

### CODE REVIEW

File: project/LibDependencies.scala [hunk 1]
```diff
@@ -11,9 +11,10 @@ object LibDependencies {
 
   val compile: Seq[ModuleID] = dependencies(
     shared = Seq(
-      "com.typesafe.play"  %% "play"            % PlayVersion.current,
-      "com.typesafe.play"  %% "filters-helpers" % PlayVersion.current,
-      "uk.gov.hmrc.webjars" % "hmrc-frontend"   % hmrcFrontendVersion
+      "org.scala-lang.modules" %% "scala-collection-compat" % "2.10.0",
```
Comment (ellamdav) [inline project/LibDependencies.scala:14]: this is needed for cross-compilation to 2.12 - it provides backports of the new collections

File: src/it/scala/uk/gov/hmrc/support/ScalaCheckUtils.scala [hunk 2]
```diff
@@ -3,25 +3,29 @@ package uk.gov.hmrc.support
 import org.scalacheck.Prop
 import org.scalacheck.Prop.collect
 
+import scala.collection.compat.immutable.LazyList
+import scala.collection.compat.immutable.LazyList.#::
```
Comment (ellamdav) [inline src/it/scala/uk/gov/hmrc/support/ScalaCheckUtils.scala:7]: Need this explicitly or else the compiler tries to use `Stream.#::` from `scala.Predef`

File: src/it/scala/uk/gov/hmrc/support/ScalaCheckUtils.scala [hunk 3]
```diff
@@ -3,25 +3,29 @@ package uk.gov.hmrc.support
...
-    case Stream.Empty       => prop
-    case h #:: Stream.Empty =>
+  def classify(conditions: LazyList[ClassifyParams])(prop: Prop): Prop = conditions match {
+    case LazyList()       => prop
+    case h #:: LazyList() =>
       if (h._1) collect(h._2)(prop) else collect(h._3)(prop)
-    case h #:: t            =>
+    case h #:: t          =>
       classify(t)(Prop.classify(h._1, h._2, h._3)(prop))
+    case _                => throw new MatchError(s"Unexpected match error on $conditions")
```
Comment (ellamdav) [inline src/it/scala/uk/gov/hmrc/support/ScalaCheckUtils.scala:29]: added this to suppress a phantom MatchError due to [a compiler bug](https://users.scala-lang.org/t/pattern-matching-lazylist/5105/10)

File: src/it/scala/uk/gov/hmrc/support/ScalaCheckUtils.scala [hunk 4]
```diff
@@ -3,25 +3,29 @@ package uk.gov.hmrc.support
...
     * @param prop       a scalacheck property [[Prop]]
     * @return [[Prop]]
     */
   @scala.annotation.tailrec
-  def classify(conditions: Stream[ClassifyParams])(prop: Prop): Prop = conditions match {
-    case Stream.Empty       => prop
-    case h #:: Stream.Empty =>
+  def classify(conditions: LazyList[ClassifyParams])(prop: Prop): Prop = conditions match {
+    case LazyList()       => prop
+    case h #:: LazyList() =>
```
Comment (ellamdav) [inline src/it/scala/uk/gov/hmrc/support/ScalaCheckUtils.scala:25]: can't use `LazyList.empty` as it's a `def`, so not a stable reference required for pattern matching

File: src/it/scala/uk/gov/hmrc/support/TemplateIntegrationBaseSpec.scala [hunk 5]
```diff
@@ -83,6 +84,8 @@ abstract class TemplateIntegrationBaseSpec[T: OWrites: Arbitrary](
             println("Skipping property evaluation")
 
             true
+          case Failure(exception)                            =>
+            throw exception
```
Comment (ellamdav) [inline src/it/scala/uk/gov/hmrc/support/TemplateIntegrationBaseSpec.scala:88]: added this to suppress exhaustiveness warning but still throw on unexpected errors

**Review by oscarduignan — APPROVED** [review]

**Pull request merged**